### ISSUE TITLE
does Batch Loan Update create one admin login per loan, or per call?

### ISSUE THREAD

**Issue by paulmoseley** [EncompassRest/EncompassRest#352]
I just found out that using the single-loan update method to update multiple loans within a service creates one Encompass 'admin login' record (along with a 'loan modified' record) for each loan updated, instead of one for the session/token (it was one per session in the SDK).  I'm seeing that I could use the Batch Update method instead, and that it's more efficient in general.  Does anyone know if Batch Update will create one Encompass 'admin login' record per batch/call, instead of one per loan?  Our users have to do monthly reconciliations of admin login usage, and I'd love to reduce the number of admin logins they need to reconcile...

**Comment by TylerBrinkley** [comment]
I'm not sure what you mean by an Encompass 'admin login' record. Could you please explain? I'd also suggest you create a separate user for making API calls instead of using the default 'admin' user. I'm not sure if the batch update process updates the loan modified date but I think it does not.

**Comment by paulmoseley** [comment]
UMPQ-PRIVATE

Hey there Tyler.  By the Encompass 'admin login' record, I'm referring to the admin user login records created by the Encompass Core system whenever the admin user is used to log into the Encompass Core system.  In Encompass Settings, go to the System Audit Trail tool (in the System Administration area), choose 'User Login' as the Category, enter 'admin' in 'Action Taken by', choose a date range and search.  Here's an example from one of our Encompass Test environments (I've also attached the screenshot):
[cid:image001.png@01D6B754.94724160]

The admin login records produced by use of the API are lumped in with other records under the Machine Name of 'Platform'.

We'd though about creating a different user specifically for API-based Encompass access, but that would just give our business owners two accounts that they'd have to monitor and report on monthly to the internal/external auditors (yes, I work for a Bank, note my e-mail address), so not sure that's worth the trouble.  By 'loan modified', I didn't mean the 'loan modified' date, but the 'loan modified' records created by the Encompass Core, and viewable in that same 'System Audit Trail' function (enter 'Loan File' in Category, and 'Modified' in Action).  I'm thinking I'll need to try out the Batch Update API endpoint myself to see what kinds of Encompass Core records it creates, but though I'd ask around too.  I'll post this response directly on the GitHub site/page, if this e-mail doesn't show up there (I'm guessing not).  Thanks for responding!

P.S. Think before you ink...join the umpqua sustainability movement and only print this email if absolutely necessary.

paul moseley
software engineer
tag - application and data engineering (ade)
mobile: 503-915-6163
Web<http://www.umpquabank.com/> | Facebook<http://www.facebook.com/umpquabank> | Twitter<http://www.twitter.com/umpquabank>

**Comment by paulmoseley** [comment]
![Encompass System Audit Trail - admin logins](https://user-images.githubusercontent.com/42451687/98726845-e49dd380-234b-11eb-9e1d-86a3e97ae7f8.PNG)

OK, I tried dragging and dropping the attachment from my previous reply, hopefully that's viewable... :)

**Comment by rbpeterson2** [comment]
Per prior comments:

  1.  Make an administrative account specific to your application or at least a generic one other than the ‘admin’ account
  2.  In your code, when running the application, are you creating a new session each time it runs the update through or only at the initialization – you should only see one login per session, either individually uploading or doing batch …

**Comment by paulmoseley** [comment]
Thanks, RB!  I'm only creating a session during initialization, not for each update.  Which is why this result surprised me.  The SDK creates one admin login per session, but the API appears to create one admin login record per use of the update patch API, even if only one session was initiated for all the loan updates requested.  Ellie Mae Tech Support has actually confirmed that this is 'expected behavior'; of course, they may have misunderstood the question :).  So I'm thinking about trying the Batch Update post API instead.

A lot of people have recommended separating out our services accesses from others by using a different Encompass account for them.  I'm considering that, but would prefer to squelch the issue a different way (by getting back to one admin login record created per session or call, instead of per update) if I can...

**Comment by paulmoseley** [comment]
OK, everyone, I tried using the Batch Loan Update function instead, and got the same result that I got for the single-loan update; the Encompass core system created one 'admin login' record for each _loan_ updated via the Batch Loan method, instead of one per _token_ (the SDK created one per _session_).  So I guess this is expected behavior :(.  I also verified that the Batch Loan Update is capable of sneaking in to a locked loan and updating it behind-the-scenes (though this may only happen after a locked loan is unlocked/exited, I'm not sure).  So Batch Loan Update is definitely useful, but not a solution to my particular 'issue'.  [BTW, TylerB, in your wiki section on Loan Updating, you've got a Batch Loan Update example that specifies 'Fields.CX.ABC' as a valid FieldValuePair field name; I discovered that, for custom fields, dropping the 'Fields.' prepend part works.  I think you explained that to me in an answer to another question I had, probably need to update your example (?).]

Several people here and at the company I work for have mentioned that I could deal with this by using a different Encompass account instead of the admin account for our API-based services.  I brought that up to the business owner, and he was delighted.  Apparently the auditors are only worried about excessive use of the admin account, NOT about other accounts with admin or super admin access to Encompass :).  So we'll be doing that instead.  Thanks!
